src/main/java/gov/nj/innovation/customAwsIdp/SamlGenerator.java

```diff
@@ -0,0 +1,228 @@
+        this.roleName = roleName;
+        this.duration = duration;
+        this.keys = keys;
+    }
+
+    /**
+     * Put together the two methods taken from {@link SamlProtocol}, this class's {@link #createUnsignedSamlResponse()} and {@link
+     * #signAndEncode(Document)}, in order to create a full SAML Response. Although this came from Keycloak, it is
+     * now purpose-built to generate one SAML Response with one Role as well as the attributes necessary to federate a
+     * sign-on to AWS Connect.
+     *
+     * @return The Base64-encoded, signed SAML Response.
+     */
+    public String getBase64SamlResponse() {
+        Document samlDocument = createUnsignedSamlResponse();
+        return signAndEncode(samlDocument);
+    }
+
+    /**
+     * This has copied the logic almost entirely from {@link SamlProtocol#authenticated(AuthenticationSessionModel,
+     * UserSessionModel, ClientSessionContext)} while cutting things down as much as possible and removing dependencies
+     * on sessions, users, clients, and key stores; instead providing instance variables and constants.
+     * <p>
+     * Unused chunks of logic have been removed, unnecessary complexity for other use-cases has been simplified where
+     * feasible.
+     *
+     * @return The unsigned XML {@link Document} which will become the SAML Response.
+     */
+    @VisibleForTesting
+    Document createUnsignedSamlResponse() {
+        SAML2LoginResponseBuilder builder = new SAML2LoginResponseBuilder();
+        builder.requestID(null)
+                .destination(REDIRECT_URI)
+                .issuer(ISSUER)
+                .assertionExpiration(ASSERTION_EXPIRATION)
+                .subjectExpiration(SUBJECT_EXPIRATION)
+                .sessionExpiration(SESSION_EXPIRATION)
+                .requestIssuer(CLIENT_ID)
+                .authMethod(JBossSAMLURIConstants.AC_UNSPECIFIED.get())
+                .includeOneTimeUseCondition(false);
+
+        List<SamlProtocol.ProtocolMapperProcessor<SAMLAttributeStatementMapper>> attributeStatementMappers =
+                new LinkedList<>();
+        AtomicReference<SamlProtocol.ProtocolMapperProcessor<SAMLRoleListMapper>> roleListMapper =
+                new AtomicReference<>(null);
+
+        ProtocolMapperModel roleMapping = RoleListMapper
+                .create("Session Role", SESSION_ROLE_ATTR_NAME, "Basic", null, true);
+        roleListMapper.set(new SamlProtocol.ProtocolMapperProcessor<>(new RoleListMapper(), roleMapping));
+
+        ProtocolMapperModel nameMapping = HardcodedAttributeMapper
+                .create("Session Name", SESSION_NAME_ATTR_NAME, "Basic", null, user);
+        attributeStatementMappers
+                .add(new SamlProtocol.ProtocolMapperProcessor<>(new HardcodedAttributeMapper(), nameMapping));
+
+        ProtocolMapperModel durationMapping = HardcodedAttributeMapper
+                .create("Session Duration", SESSION_DURATION_ATTR_NAME, "Basic", null, duration);
```

Comment:
nit: when I see any String in code in Java I think "oh that could be a constant", but it's not always important. What do you think for `Session Duration` and `Basic`? (or is this a case of the Keycloak code just being like that already?)

Reply:
Yeah... I was borderline on these ones, since they are used very sparingly. I'll just make them constants.